### PR TITLE
[software] PanoramaInit: Add new restriction to circle detection

### PR DESCRIPTION
**Pull request by servantf**
Add some new restriction to circle detection

### CODE REVIEW

File: src/software/pipeline/main_panoramaInit.cpp [hunk 1]
```diff
@@ -216,6 +216,8 @@ class CircleDetector {
     double level_centerx = _center_x / pow(2.0, level);
     double level_centery = _center_y / pow(2.0, level);
     double level_radius = _radius / pow(2.0, level);
+    int min_radius = gradients.Width() / 2;
```
Comment (fabiencastan) [inline src/software/pipeline/main_panoramaInit.cpp:219]: To be independant from the image orientation:
```suggestion
    int min_radius = std::min(gradients.Width(), gradients.Height()) / 2;
```

File: src/software/pipeline/main_panoramaInit.cpp [hunk 2]
```diff
@@ -229,7 +231,7 @@ class CircleDetector {
       int max_x = -1;
 
       /*Lookup possible radius*/
-      int start = std::max(0, int(level_radius) - uncertainty);
+      int start = std::max(min_radius, int(level_radius) - uncertainty);
```
Comment (fabiencastan) [inline src/software/pipeline/main_panoramaInit.cpp:234]: Could we also add a way to not be bigger than the diagonal?

Comment (fabiencastan) [reply src/software/pipeline/main_panoramaInit.cpp:234]: and a min/max also for the optical center like in the center within 20% of the image.

File: src/software/pipeline/main_panoramaInit.cpp (hunk 1 above)
Comment (servantf) [reply src/software/pipeline/main_panoramaInit.cpp:219]: No ! Gradients is a polar image !

File: src/software/pipeline/main_panoramaInit.cpp (hunk 2 above)
Comment (servantf) [reply src/software/pipeline/main_panoramaInit.cpp:234]: The diagonal limit is implicit thanks to polar image construction

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**